packages/internals/support/src/component.js

```diff
@@ -1,30 +1,73 @@
 const assert = require('check-types').assert;
 const check = require('check-more-types');
+const {omit} = require('lodash');
 const Entity = require('./entity');
 const Collection = require('./collection');
 const File = require('./file');
 const FileCollection = require('./file-collection');
 
+const isFile = value => value instanceof File;
+check.mixin(isFile, 'file');
+const isCollection = value => value instanceof Collection;
+check.mixin(isCollection, 'collection');
+const isFileCollection = value => value instanceof FileCollection;
+check.mixin(isFileCollection, 'fileCollection');
+
```

Comment:
I like this... could be useful to have our own pre-customised `check` instance which has these Fractal specific tests already mixed in as a shared resource somewhere. Maybe as part or utils/supports? A task for later maybe...